index.d.ts

```diff
@@ -291,102 +291,52 @@ export interface KeyInfoProvider {
    * @param keyInfo - An optional array of XML Nodes.
    * @return A string or Buffer representing the key.
    */
-  getKey(keyInfo?: Node[]): string | Buffer;
+  getCert(keyInfo?: Node[]): string | Buffer;
 
   /**
    * Method to return an XML string representing the contents of a KeyInfo element.
    *
-   * @param key - An optional string representing the key.
+   * @param cert - An optional string, or array of strings, representing the certificate to be included in the KeyInfo element.
    * @param prefix - An optional string representing the namespace alias.
    * @return An XML string representation of the contents of a KeyInfo element.
    */
-  getKeyInfo(key?: string, prefix?: string): string;
+  makeKeyInfo(cert?: string | string[], prefix?: string): string;
 
   /**
    * An optional dictionary of attributes which will be added to the KeyInfo element.
    */
-  attrs?: { [key: string]: string };
+  attrs?: { [cert: string]: string };
```

Comment:
```suggestion
  attrs?: { [key: string]: string };
```

Reply:
But this isn't a key to be provided here, it is a cert as it is part of the `<X509Certificate />` payload. Am I missing something?

Reply:
The `key ` part of the index signature `[key: string]` is just a placeholder value, typically people just put `key`,
If you want to be more precise you could put something like `attrName` there.
And the values of attr are not part of the X509Certificate payload, they are attributes of the KeyInfo element.
For example if  `attr = {Id:"12345", foo:"bar"}` then `<KeyInfo Id="12345" foo="bar">`

Reply:
Right... a TypeScript thing... that is a crazy oversight on my part. Too much language hopping lately :)